src/proxy/processors/push-action/parsePush.ts

```diff
@@ -41,8 +35,10 @@ async function exec(req: any, action: Action): Promise<Action> {
       action.addStep(step);
       return action;
     }
+    console.log('RRR');
     const [packetLines, packDataOffset] = parsePacketLines(req.body);
-    const refUpdates = packetLines.filter((line) => line.includes(BRANCH_PREFIX));
+    console.log('PPPP');
+    const refUpdates = packetLines.filter((line) => line.includes('refs/'));
 
     if (refUpdates.length !== 1) {
```

Comment:
The flow described in the issue seems to be failing here, due to "multiple ref updates":

```bash
git tag --annotate foo -m ''
git cat-file tag foo
git push --tags
```

<img width="1127" height="166" alt="Image" src="https://github.com/user-attachments/assets/26016e61-f61d-4b04-a765-00cea071e194" />

I think this is happening due to tags (or other objects) that haven't been approved yet, and git is trying to push everything at once when calling `git push --tags`.

This will probably be complicated to solve - for the time being we could improve the error message to say something like:

```ts
step.setError(
  'Your push has been blocked. Multi-ref pushes (multiple tags and/or branches) are not supported yet. Please push one ref at a time.',
);
```

Reply:
Note that when I approved my previous tag push, it did reduce the number of refs, so it's just bundling and pushing all the unapproved branches + tags at once.

<img width="1127" height="166" alt="Image" src="https://github.com/user-attachments/assets/1eecb617-8fa2-40c9-a906-33476bc2eb89" />